components/toolbar/ToolbarButton.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { HTMLAttributes } from 'react';
+import { classnames } from '../../helpers/component';
+import Icon, { Props as IconProps } from '../icon/Icon';
+
+interface Props extends HTMLAttributes<HTMLButtonElement> {
+    icon: string | IconProps;
+}
+
+const ToolbarButton = ({ icon, className, ...rest }: Props) => {
+    return (
+        <button type="button" className={classnames([className, 'toolbar-button'])} {...rest}>
+            {typeof icon === 'string' ? (
+                <Icon name={icon} className="toolbar-icon mauto" />
+            ) : (
+                <Icon {...icon} className={classnames([icon.className, 'toolbar-icon mauto'])} />
```

Comment:
We ignore `rest` on purpose here?

Reply:
What do you mean? Rest is passed to button element as intended. Icon can be a string - icon name (should be good enough for most cases), or an object - props for icon, when something custom is needed, like image rotation, or different fill colors.

Reply:
Oh, crap, I don't know why I saw `rest` passed in `<Icon>` and not in `<button>`... :man_facepalming: